pkg/cloud/azure/actuators/machine_scope.go

```diff
@@ -191,3 +211,80 @@ func unmarshalProviderSpec(spec *runtime.RawExtension) (*v1alpha1.AzureMachinePr
 	klog.V(6).Infof("Found ProviderSpec: %+v", config)
 	return &config, nil
 }
+
+func updateScope(coreClient controllerclient.Client, credentialsSecret *apicorev1.SecretReference, scope *Scope) error {
+	if credentialsSecret == nil {
+		return errors.New("provided empty credentials secret")
+	}
+
+	secretName := fmt.Sprintf("%s/%s", credentialsSecret.Namespace, credentialsSecret.Name)
```

Comment:
This is not a big deal, but just a heads up: It looks like this is used for log messages. `ObjectKey` is just a type alias for `types.NamespacedName` which has a `String()` method, so you can just use that directly in log lines.

Reply:
Nice! changed.